[Nyamdavaa Amar] iP

nvmdava opened the pull request:
#Duke
- Bullet
- List
1. Numbered
2. List
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
- [x] Done
- [ ] Not Done
💯💯💯
`block quote`
[hyperlink ](https://github.com/nus-cs2103-AY2223S1/ip/pull/459)
`Inline code == true`
~~strikethrough~~
_italic_
**bold**

Commit:
Add Gradle support

nvmdava committed:
Adding basic Greet, Bye and Echo functions

nvmdava committed:
Added a list of tasks, method to list or add them

nvmdava committed:
added a way to mark the tasks done/undone

nvmdava committed:
tasks are now split into todo/event/deadline types

nvmdava committed:
added basic testcase

nvmdava committed:
error handling

nvmdava committed:
delete task feature

nvmdava committed:
Add way to save data to file

nvmdava committed:
Add LocalDate to save date

nvmdava committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Task.java
#	text-ui-test/EXPECTED.TXT
#	text-ui-test/input.txt

nvmdava committed:
Add parser, tasklist objects and command interface

nvmdava committed:
Wrap Duke into package

nvmdava committed:
Merge remote-tracking branch 'upstream/add-gradle-support'
Add gradle support

nvmdava committed:
Add JUnit tests

nvmdava committed:
Add Javadoc comments

nvmdava committed:
Do checkstyle

nvmdava committed:
Add find command to search tasks

nvmdava committed:
Merge branch 'branch-A-JavaDoc'

nvmdava committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/DukeException.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Task.java
#	src/main/java/duke/TaskList.java

nvmdava committed:
Add varargs to find method

WR3nd3 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,8 @@
+ * Command interface to be passed onto Parser.
+ */
+interface Command {
+    void execute(String string) throws DukeException;
```
Perhaps you can write a javadoc comment for this execute method and {@inherit} the overriding methods?

WR3nd3 replied:
Similarly, javadoc headers are missing for most public classes/methods. I believe those should be completed.

WR3nd3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,130 @@
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+
+        UserInterface.print("Hello from\n" + logo);
```
Maybe it would be better abstraction if all the code/methods constructing messages to print is also encapsulated into the UserInterface class instead of only using it to hold the print method.

WR3nd3 replied:
This can apply to all parts of the code that use UserInterface.print(String) since it appears very frequently.

WR3nd3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,130 @@
+     */
+    private void fillParser() {
+        parser.addCommand("todo", argument -> {
+            if (argument.equals("")) {
```
The String methods isBlank() and isEmpty() could replace the equals("")

WR3nd3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,130 @@
+            }
+            Task item = Task.toDo(argument);
+            taskList.addTask(item);
+            UserInterface.print("added: " + item);
```
Maybe it would be better abstraction if all the code/methods constructing messages to print is also encapsulated into the UserInterface class instead of only using it to hold the print method. This would also be useful to make the getResponse() method for the upcoming iP tasks simpler to implement.

WR3nd3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,130 @@
+    /**
+     * Initialize Parser with built-in commands.
+     */
+    private void fillParser() {
```
Would it be better to define the behavior of the commands in the Parser class instead of defining them in the Duke class?

WR3nd3 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,128 @@
+    private String description;
+    private boolean isDone = false;
+
+    private static class ToDo extends Task {
```
Would it be better to define the subclasses of Task in their own class files?

gowribhat commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,130 @@
+            taskList.addTask(item);
+            UserInterface.print("added: " + item);
+            UserInterface.print("Now you have " + taskList.getSize() + " tasks in the list.");
+        });
```
Do you think code duplication can be avoided here? Perhaps a helper function can be used.

gowribhat commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,128 @@
+            throw DukeException.INVALIDARGUMENT;
+        }
+        return new Event(arr[0], arr[1]);
+    }
```
Perhaps the method names can be made clearer to show what it does in verb form (eg 'createDeadline') 😄

gowribhat commented on src/test/java/duke/TaskTest.java:
```diff
@@ -0,0 +1,32 @@
+
+public class TaskTest {
+    @Test
+    public void TaskTest() {
```
Maybe you could have separate tests instead of one test with multiple assertions?

gowribhat commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,83 @@
+     * @return the task at position.
+     * @throws DukeException when the task position is non-existent.
+     */
+    public Task getTask(int i) throws DukeException {
```
Would it be possible to abstract this out? You could potentially have methods to mark/unmark in `TaskList` class so as to avoid having to pass `Task` objects around 😄

gowribhat commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,83 @@
+ */
+class TaskList {
+
+    private ArrayList<Task> listOfTasks = new ArrayList<>();
```
Good naming through the use of plural!

gowribhat reviewed (commented):
Overall, code quality guidelines have been followed well. Only some minor changes are needed 😄

nvmdava committed:
Add GUI

Currently the user uses CLI to interact with Duke, which is inconvenient for non-tech people.
Therefore we added GUI to Duke and fully deprecated the CLI.

Every command that was in CLI is same in GUI.

nvmdava committed:
Improve Code Quality

Improved code quality for better readability.

nvmdava committed:
Add asserts

Added asserts for additional checks

nvmdava committed:
Add help command

User may forget the commands, therefore a default help command should be implemented.

nvmdava committed:
Add User Guide

nvmdava committed:
Delete Ui.PNG

nvmdava committed:
Add files via upload

nvmdava committed:
Rename file

nvmdava committed:
Merge branch 'master' of https://github.com/nvmdava/ip
